react/features/app/actions.js

```diff
@@ -31,176 +31,70 @@ declare var APP: Object;
-            location.port = defaultLocation.port;
-            location.protocol = defaultLocation.protocol;
-        } else {
-            // eslint-disable-next-line no-param-reassign
-            location = defaultLocation;
+    return async(dispatch: Dispatch<any>, getState: Function) => {
+        let location = parseURIString(uri);
+
+        // If the specified location (URI) does not identify a host, use the app's
+        // default.
+        if (!location || !location.host) {
+            const defaultLocation = parseURIString(getDefaultURL(getState));
+
+            if (location) {
+                location.host = defaultLocation.host;
+
+                // FIXME Turn location's host, hostname, and port properties into
+                // setters in order to reduce the risks of inconsistent state.
+                location.hostname = defaultLocation.hostname;
+                location.pathname
+                    = defaultLocation.pathname + location.pathname.substr(1);
+                location.port = defaultLocation.port;
+                location.protocol = defaultLocation.protocol;
+            } else {
+                // eslint-disable-next-line no-param-reassign
+                location = defaultLocation;
+            }
         }
-    }
-
-    location.protocol || (location.protocol = 'https:');
-
-    return _appNavigateToMandatoryLocation(dispatch, getState, location);
-}
 
-/**
- * Loads config.js from a specific host.
- *
- * @param {Dispatch} dispatch - The redux {@code dispatch} function.
- * @param {Function} getState - The redux {@code getState} function.
- * @param {Object} location - The location URI which specifies the host to load
- * the config.js from.
- * @private
- * @returns {Promise<Object>}
- */
-function _loadConfig(
-        dispatch: Dispatch<any>,
-        getState: Function,
-        { contextRoot, host, protocol, room }) {
-    // XXX As the mobile/React Native app does not employ config on the
-    // WelcomePage, do not download config.js from the deployment when
-    // navigating to the WelcomePage - the perceived/visible navigation will be
-    // faster.
-    if (!room && typeof APP === 'undefined') {
-        return Promise.resolve();
-    }
+        location.protocol || (location.protocol = 'https:');
+        const locationURL = new URL(location.toString());
 
-    /* eslint-disable no-param-reassign */
+        dispatch(configWillLoad(locationURL));
 
-    protocol = protocol.toLowerCase();
+        let protocol = location.protocol.toLowerCase();
+        const { contextRoot, host, room } = location;
 
-    // The React Native app supports an app-specific scheme which is sure to not
-    // be supported by fetch (or whatever loadConfig utilizes).
-    protocol !== 'http:' && protocol !== 'https:' && (protocol = 'https:');
+        // The React Native app supports an app-specific scheme which is sure to not
+        // be supported by fetch (or whatever loadConfig utilizes).
```

Comment:
can you remove the `or whatever loadConfig utilizes` part? It doesn't add anything.